libherokubuildpack/src/buildpack_output/mod.rs

```diff
@@ -491,6 +524,40 @@ where
         }
     }
 
+    /// Output periodic timer updates to the end user.
+    ///
+    /// If a buildpack author wishes to start a long-running task that does not stream, starting a timer
+    /// will let the user know that the buildpack is performing work and that the UI is not stuck.
+    ///
+    /// One common use case is when downloading a file. Emitting periodic output when downloading is especially important for the local
+    /// buildpack development experience where the user's network may be unexpectedly slow, such as
+    /// in a hotel or on a plane.
+    ///
+    /// This function will transition your buildpack output to [`state::Background`].
+    #[allow(clippy::missing_panics_doc)]
+    pub fn start_timer(mut self, s: impl AsRef<str>) -> BuildpackOutput<state::Background<W>> {
+        // Do not emit a newline after the message
+        write!(self.state.write, "{}", Self::style(s)).expect("Output error: UI writer closed");
+        self.state
+            .write
+            .flush()
+            .expect("Output error: UI writer closed");
+
+        BuildpackOutput {
+            started: self.started,
+            state: state::Background {
+                started: Instant::now(),
+                write: background_printer::print_interval(
+                    self.state.write,
+                    std::time::Duration::from_secs(1),
+                    ansi_escape::wrap_ansi_escape_each_line(&ANSI::Dim, " ."),
+                    ansi_escape::wrap_ansi_escape_each_line(&ANSI::Dim, "."),
+                    ansi_escape::wrap_ansi_escape_each_line(&ANSI::Dim, ". "),
```

Comment:
For dimming the dots:

The downside of this approach is that each character is wrapped in color codes. The upside is we never have to worry about accidentally getting in a state where the output is dimmed, and a panic happens, and the RESET is never written by accident.

Reply:
I think the upside is worth it.

Reply:
The more I think about it the more I like the strong guarantee to not put our user's terminal in a "bad" state with an ANSI code.